crates/ruma-common/src/push/predefined.rs

```diff
@@ -631,6 +640,13 @@ pub enum PredefinedOverrideRuleId {
     _Custom(PrivOwnedStr),
 }
 
+impl PredefinedOverrideRuleId {
+    /// Get the kind of this `PredefinedOverrideRuleId`.
+    pub fn kind(&self) -> RuleKind {
+        RuleKind::Override
+    }
+}
```

Comment:
Are these methods really useful? I get `PredefinedRuleId::kind` which actually does a match, but the methods for `Predefined...RuleId` seem kind of pointless, no?

Reply:
They are for lazy programmers like me :stuck_out_tongue: . The idea is to be able to do:

```rust
let rule_id = PredefinedOverrideRuleId::Master;
ruma::api::client::push::set_pushrule_enabled::v3::Request::enable(RuleScope::Global, rule_id.kind(), rule_id);
```

i.e. you don't have to provide the `RuleKind` explicitely when it is known because of the type.

Reply:
Sounds like another constructor for `v3::Request` would be useful?

Reply:
It might, although we would need them for all methods that take a rule ID, i.e. ~ half of the `Request` constructors in the `api::client::push` module, and also probably some methods on [`Ruleset`](https://docs.ruma.io/ruma/push/struct.Ruleset.html), like `get()`. And maybe other places I'm not thinking about right now.

Reply:
Although to be honest, using the SDK I would probably not use these methods, but it still has methods that take the kind and rule ID.